[CI] SearchCancellationIT testCancelFailedSearchWhenPartialResultDisallowed failing

**Build Scans:**
- [elasticsearch-periodic #6564 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/r3ateajvomuds)
- [elasticsearch-pull-request #58782 / part-1](https://gradle-enterprise.elastic.co/s/66toybp2nrj2c)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.search.SearchCancellationIT.testCancelFailedSearchWhenPartialResultDisallowed" -Dtests.seed=B60636170B09C981 -Dtests.locale=uk -Dtests.timezone=Pacific/Truk -Druntime.java=17 -Dtests.fips.enabled=true
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.SearchCancellationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testCancelFailedSearchWhenPartialResultDisallowed),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: The Coordinator should have one SearchTask.
Expected: a collection with size <1>
     but: collection size was <0>
```

**Issue Reasons:**
- [8.x] 2 failures in test testCancelFailedSearchWhenPartialResultDisallowed (1.5% fail rate in 132 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java
@@ -262,13 +262,17 @@ public class SearchCancellationIT extends AbstractSearchCancellationTestCase {
         // When the search request executes, block all shards except 1.
         final List<SearchShardBlockingPlugin> searchShardBlockingPlugins = initSearchShardBlockingPlugin();
         AtomicBoolean letOneShardProceed = new AtomicBoolean();
+        // Ensure we have at least one task waiting on the latch
+        CountDownLatch waitingTaskLatch = new CountDownLatch(1);
         CountDownLatch shardTaskLatch = new CountDownLatch(1);
         for (SearchShardBlockingPlugin plugin : searchShardBlockingPlugins) {
             plugin.setRunOnNewReaderContext((ReaderContext c) -> {
                 if (letOneShardProceed.compareAndSet(false, true)) {
                     // Let one shard continue.
                 } else {
-                    safeAwait(shardTaskLatch, TimeValue.timeValueSeconds(30)); // Block the other shards.
+                    // Signal that we have a task waiting on the latch
+                    waitingTaskLatch.countDown();
+                    safeAwait(shardTaskLatch); // Block the other shards.
                 }
             });
         }
@@ -280,6 +284,9 @@ public class SearchCancellationIT extends AbstractSearchCancellationTestCase {
             plugin.disableBlock();
             plugin.setBeforeExecution(() -> {
                 if (oneThreadWillError.compareAndSet(false, true)) {
+                    // wait for some task to get to the latch
+                    safeAwait(waitingTaskLatch);
+                    // then throw the exception
                     throw new IllegalStateException("This will cancel the ContextIndexSearcher.search task");
                 }
             });

